src/main/java/io/github/mirromutth/r2dbc/mysql/internal/MySqlSession.java

```diff
@@ -48,37 +44,11 @@
      */
     private final CharCollation collation = CharCollation.clientCharCollation();
 
-    private volatile int clientCapabilities = 0;
-
-    /**
-     * It would be null after connection phase completed.
-     */
-    @Nullable
-    private volatile MySqlAuthProvider authProvider;
+    private volatile int capabilities = 0;
 
-    /**
-     * It would be null after connection phase completed.
-     */
-    @Nullable
-    private volatile String username;
```

Comment:
All things about authentication are moved to `LoginFlow`, stay `MySqlSession` clean.